src/main/java/com/bootcamp/demo/Controller/AccountController.java

```diff
@@ -31,6 +31,30 @@ void add(@RequestBody Account account) throws ControllerException {
         }
     }
 
+    @PutMapping("/editAccount")
+    void update(@RequestBody AccountDetails accountDetails) throws ControllerException {
+        System.out.println("update accountDetails-->" + accountDetails.getName());
+        try {
+            Account searchedAccount = service.findOne(accountDetails.getUsername());
+
+            if (searchedAccount != null) {
+                Account account = new Account(
+                        accountDetails.getName(),
+                        accountDetails.getEmail(),
+                        accountDetails.getUsername(),
+                        searchedAccount.getPassword(),
+                        accountDetails.getPhoneNumber(),
+                        accountDetails.getAddress(),
+                        accountDetails.getDateOfBirth()
+                );
+                this.service.update(account);
+            }
+
+        } catch (ServiceException | ExecutionException | InterruptedException e) {
+            throw new ControllerException(e.getMessage());
+        }
```

Comment:
This exception isn't caught anywhere, so I think you should return a response body instead, along with an appropriate http status.